tlsutil/config.go

```diff
@@ -87,6 +89,8 @@ type Config struct {
 	// PreferServerCipherSuites specifies whether to prefer the server's ciphersuite
 	// over the client ciphersuites.
 	PreferServerCipherSuites bool
+
+	EnableAgentTLSForChecks bool
```

Comment:
Is this a new feature?

Reply:
This is an existing feature. In order to be able to generate the correct `tls.config` for agent checks, `tls.Configurator` needs to know if this flag is set or not. https://www.consul.io/docs/agent/options.html#enable_agent_tls_for_checks.